### ISSUE TITLE
Git authentication fails

### ISSUE THREAD

**Issue by bripkens** [gitbutlerapp/gitbutler#2582]
Hey folks,

the git authentication is failing for me through the Gitbutler client. It is however working fine through the `git` CLI and various IDEs (Intellij and VSCode for example).

My setup:

 - GitHub auth via SSH keys
 - The SSH identity is stored in my OpenSSH auth agent (`ssh-add -l`)
 - The identity to use for GitHub is defined within my SSH config (`cat ~/.ssh/config`)

Within Gitbutler, I tried the `Auto Detect` mode, as well as `Use existing SSH key` (without passphrase).

Error shown within Gitbutler:

<img width="420" alt="image" src="https://github.com/gitbutlerapp/gitbutler/assets/596443/ba7793e4-ab44-4a30-9589-53c2feb6f7ff">

**Comment by mtsgrd** [comment]
Hey, sorry about the delay and about the issue you're running into. @bripkens do you use a git remote with an invalid host in order to specify the identity file? This is currently known to break the app, but we're keen to find a solution.

**Comment by bripkens** [comment]
> @Hey, sorry about the delay and about the issue you're running into. @bripkens do you use a git remote with an invalid host in order to specify the identity file? This is currently known to break the app, but we're keen to find a solution.

To the best of my knowledge nothing is misconfigured/has an invalid host. For your info, here are excerpts of the files. And as mentioned, the config is working with all other solutions I am using (VSCode, Intellij and the git client itself):

```
$ cat ~/.ssh/config
Host github.com
  IdentityFile ~/.ssh/ed25519%
$ cat ~/.gitconfig
[credential]
  helper = osxkeychain
[url "git@github.com:"]
  insteadOf = https://github.com/
$ ssh-add -l
256 SHA256:Ro9{elided}s9yV5k ben.blackmore@dash0.com (ED25519)
$ git remote -v
origin  git@github.com:dash0hq/{elided}.git (fetch)
origin  git@github.com:dash0hq/{elided}.git (push)
```

**Comment by Byron** [comment]
By now, there is a new option that allows to use the Git binary for fetch and push.

<img width="616" alt="Screenshot 2024-04-22 at 15 54 01" src="https://github.com/gitbutlerapp/gitbutler/assets/63622/cbcadd5f-ec7c-44ed-8aeb-71f46f751d60">

Would you be able to try this and let us know if it works? Thank you.

**Comment by bripkens** [comment]
@Byron Pushing works 🎉. However, the commit becomes unverified.

<img width="410" alt="image" src="https://github.com/gitbutlerapp/gitbutler/assets/596443/061d5a2f-927b-4dd0-be3e-3c16b26fc217">

**Comment by Byron** [comment]
That's great to hear! 

Regarding the signing issue, my assumption is that you have configured signing in Git, but GitButler commits are not necessarily respecting this configuration yet. However, it is able to sign commits itself using the generated SSH-Key, but it's a GitButler setting which might not be configured here.

**Comment by Byron** [comment]
Regarding the unsigned commit, could you share your setup and roughly what you did to get that unsigned commit? *(I think I have an idea, but want to be sure)*. From there, I'd be able to create a new issue and close this one. Thanks again for your help.

**Comment by bripkens** [comment]
Sure @Byron

> Regarding the unsigned commit, could you share your setup and roughly what you did to get that unsigned commit? 

From a UI perspective, I just went into Gitbutler and hit the commit button followed by push. I haven't configured any of Gitbutler's SSH keys/Gitbutler specific signing. 

I would personally prefer to rely on Git's native signing capabilities and configuration.

The `.gitconfig` contains the usual. Also fyi, I can push without getting prompted for a GPG key passphrase (that is, if I have pushed once so far and the GPG session hasn't expired).

```
[user]
  name = Ben Blackmore
  email = ben.blackmore@dash0.com
  signingkey = 3C89A8D52B4321F6
[credential]
  helper = osxkeychain
[commit]
  gpgsign = true
```
